### PR TITLE
List by tag

### PR DESCRIPTION
**Pull request by vincentcreusotfx**
-- Note
the import of the config package will need to be changed before asking a PR to RobustPerception

### CODE REVIEW

File: .promu.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 go:
     cgo: false
 repository:
-    path: github.com/RobustPerception/azure_metrics_exporter
+    path: github.com/FXInnovation/azure_metrics_exporter
```
Comment (louisfelix) [inline .promu.yml]: Why do you need to put "FXInnovation", even temporarily? (how do you use `.promu.yml`)

File: README.md [hunk 2]
```diff
@@ -105,6 +111,16 @@ List of regexps that is matched against the resource name.
 Metrics of all matched resources are ignored (defaults to exclude none)
 Excludes take precedence over the include filter.
 
+### Resource tag filtering
+
+Resources having a specific tag can be filtered:
```
Comment (louisfelix) [inline README.md]: Maybe add more precision: 
```suggestion
Resources having a specific tag name and tag value can be filtered:
```

File: README.md [hunk 3]
```diff
@@ -105,6 +111,16 @@ List of regexps that is matched against the resource name.
...
+
+`resource_tag_name`:
+Name of the tag to be filtered against.
+
+`resource_tag_value`:
+Value that must me matched for the tag name passed.
```
Comment (louisfelix) [inline README.md]: Maybe more simply:
```suggestion
Value of the tag to be filtered against.
```

File: azure.go [hunk 4]
```diff
@@ -3,14 +3,13 @@ package main
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/FXInnovation/azure_metrics_exporter/config"
```
Comment (louisfelix) [inline azure.go]: As you said in the "note": to be fixed

File: azure.go [hunk 5]
```diff
@@ -279,32 +332,29 @@ func (ac *AzureClient) listFromResourceGroup(resourceGroup string, resourceTypes
...
+	return body, err
 
-	var data AzureResourceListResponse
-	err = json.Unmarshal(body, &data)
-	if err != nil {
-		return nil, fmt.Errorf("Error unmarshalling response body: %v", err)
-	}
+}
 
+func extractMetricNames(data AzureResourceListResponse, subscription string) []string {
```
Comment (louisfelix) [inline azure.go]: Are you sure you didn't mean:
```suggestion
// Extract resource names from the AzureResourceListResponse
func extractResourceNames(data AzureResourceListResponse, subscription string) []string {
```

File: azure.go [hunk 6]
```diff
@@ -279,32 +332,29 @@ func (ac *AzureClient) listFromResourceGroup(resourceGroup string, resourceTypes
 		return nil, fmt.Errorf("Error: %v", err)
 	}
 	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+
 	if resp.StatusCode != 200 {
-		return nil, fmt.Errorf("Unable to query resource group API with status code: %d", resp.StatusCode)
+		return nil, fmt.Errorf("Unable to query resource API with status code: %d with body: %s", resp.StatusCode, body)
```
Comment (louisfelix) [inline azure.go]: Should be (I think):
```suggestion
		return nil, fmt.Errorf("Unable to query API with status code: %d and with body: %s", resp.StatusCode, body)
```

File: azure.go [hunk 7]
```diff
@@ -268,7 +276,52 @@ func (ac *AzureClient) listFromResourceGroup(resourceGroup string, resourceTypes
 
 	metricValueEndpoint := fmt.Sprintf("https://management.azure.com/%s/resourceGroups/%s/resources?api-version=%s&$filter=%s", subscription, resourceGroup, apiVersion, filterTypes)
```
Comment (louisfelix) [inline azure.go]: Variable should be `resourcesEndpoint` here.

File: azure.go [hunk 8]
```diff
@@ -268,7 +276,52 @@ func (ac *AzureClient) listFromResourceGroup(resourceGroup string, resourceTypes
...
+
+// Returns all resource with the given couple tagname, tagvalue
+func (ac *AzureClient) listByTag(tagName string, tagValue string) ([]string, error) {
+	apiVersion := "2018-05-01"
+
+	filterTypes := url.QueryEscape(fmt.Sprintf("tagName eq '%s' and tagValue eq '%s'", tagName, tagValue))
+
+	subscription := fmt.Sprintf("subscriptions/%s", sc.C.Credentials.SubscriptionID)
+
+	metricValueEndpoint := fmt.Sprintf("https://management.azure.com/%s/resources?api-version=%s&$filter=%s", subscription, apiVersion, filterTypes)
```
Comment (louisfelix) [inline azure.go]: Variable should be `resourcesEndpoint` here.

File: main.go [hunk 9]
```diff
@@ -1,13 +1,13 @@
 package main
 
 import (
+	"github.com/FXInnovation/azure_metrics_exporter/config"
```
Comment (louisfelix) [inline main.go]: As you said in the "notes", to be fixed

File: azure.go (hunk 4 above)
Comment (vincentcreusotfx) [reply azure.go]: I see a "workaround" but it's pretty annoying. 
1/ do a PR with only the config change
2/ do the 2nd PR ...

File: azure.go (hunk 7 above)
Comment (vincentcreusotfx) [reply azure.go]: Right

File: azure.go (hunk 8 above)
Comment (vincentcreusotfx) [reply azure.go]: Yep

File: .promu.yml (hunk 1 above)
Comment (vincentcreusotfx) [reply .promu.yml]: Looking at it ... will give you some explanations

Comment (vincentcreusotfx) [reply .promu.yml]: If I don't change it, it will go build the RobustPerception version, not mine

File: azure.go (hunk 7 above)
Comment (louisfelix) [reply azure.go]: @vincentcreusotfx This one is not fixed

Comment (vincentcreusotfx) [reply azure.go]: Done it ... it was not my code so I let it like it was first

File: .promu.yml (hunk 1 above)
Comment (louisfelix) [reply .promu.yml]: See:
http://blog.campoy.cat/2014/03/github-and-go-forking-pull-requests-and.html
http://code.openark.org/blog/development/forking-golang-repositories-on-github-and-managing-the-import-path
https://www.reddit.com/r/golang/comments/2jdcw1/how_do_you_deal_with_github_forking/

File: azure.go (hunk 4 above)
Comment (louisfelix) [reply azure.go]: See:
http://blog.campoy.cat/2014/03/github-and-go-forking-pull-requests-and.html
http://code.openark.org/blog/development/forking-golang-repositories-on-github-and-managing-the-import-path
https://www.reddit.com/r/golang/comments/2jdcw1/how_do_you_deal_with_github_forking/

File: main.go (hunk 9 above)
Comment (louisfelix) [reply main.go]: Fixed.

**Review by louisfelix — APPROVED** [review]

**Review by louisfelix — APPROVED** [review]

**Review by ZakiChammaa — APPROVED** [review]

**Review by thomaslacher — APPROVED** [review]

**Comment by vincentcreusotfx** [comment]
Fixed the import problem by using a new remote on the original before fork repo locally, so first note on PR is fixed

**Pull request merged**